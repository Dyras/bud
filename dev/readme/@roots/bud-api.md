## Overview

Surfaces various configuration utilities for use in end-user configs and extensions.

[See the documentation](https://bud.js.org/docs/api.alias)
