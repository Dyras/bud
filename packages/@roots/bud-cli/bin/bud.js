#!/usr/bin/env node

const CLI = require('../lib/cjs/CLI').CLI
const commands = require('../lib/cjs/CLI/commands')

new CLI({
  command: 'bud',
  projectUrl: 'https://github.com/roots/bud',
  commands,
})
  .mast()
  .invoke()
