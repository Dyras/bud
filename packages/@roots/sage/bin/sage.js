#!/usr/bin/env node

const {sage: app} = require('../lib/cjs')
const Bud = require('@roots/bud-cli')

new Bud.CLI({
  name: 'sage',
  app,
  projectUrl: 'https://github.com/roots/sage',
  commands: Bud.commands,
}).invoke()
