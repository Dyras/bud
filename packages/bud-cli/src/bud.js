#!/usr/bin/env node
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bud = void 0;
const bud_support_1 = require("@roots/bud-support");
const build = __importStar(require("./commands/build"));
const publish = __importStar(require("./commands/publish"));
/**
 * $bud
 */
const cli = bud_support_1.yargs
    .command(build)
    .command(publish)
    .recommendCommands()
    .demandCommand(1, 'You must specify a command. See `bud --help` for usage.\n')
    .usage('\nBud-CLI \n\nbud [command] [options]')
    .version()
    .wrap(bud_support_1.yargs.terminalWidth())
    .epilog('https://github.com/roots/bud').argv;
exports.bud = cli;
//# sourceMappingURL=bud.js.map