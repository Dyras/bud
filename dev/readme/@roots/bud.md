## Installation

Install [@roots/bud](/packages/@roots/bud) and [**@roots/bud-cli**](/packages/@roots/bud-cli) to your project

```sh
yarn add @roots/bud @roots/bud-cli --dev
```

## Getting started

Bud can either be configured with a static config file (`json`/`yml`) or a builder module (`js`/`ts`).

Dead simple example:

```js
/**
 * bud.config.js
 */
module.exports = (bud) => bud.entry('app', ['app.js']);
```

Or, as yml:

```yml
# bud.config.yml
entry:
  app: 'app.js'
```

A more advanced configuration might look like:

```ts
/**
 * bud.config.ts
 */
import { Framework } from "@roots/bud";

export default (bud: Framework) =>
  bud
    .use([
      require('@roots/bud-babel'),
      require('@roots/bud-postcss'),
      require('@roots/bud-react'),
    ])
    .setPath('src', 'js')
    .setPublicPath('assets')
    .alias({
      '@src': bud.path('src'),
    })

    .entry({ app: 'app.{js,css}' })

    .when(
      bud.isProduction,
      ({ minimize }) => minimize(),
      ({ devtool }) => devtool()
    );
```

For more on configuring [**@roots/bud**](/packages/@roots/bud) check out the [dedicated documentation](/docs/config/README.md).

## Running a build

![Terminal usage](https://github.com/roots/bud/tree/stable/static/img/cli.svg)

Once you've set up your configuration file the following command will run the build:

```sh
yarn bud build
```

You should see your built assets in the `dist` directory of your project.

### Running in `production` mode

```sh
yarn bud build:production
```

### Running in `development` mode

```sh
yarn bud build:development
```

## Example implementations

There are [example implementations available](/examples).

## Extending

Bud provides an intentionally sparse set of out-of-the-box features.

In fact, much of the core of Bud is actually made up of extensions. This is to make it easy for devs to swap out parts of the framework as needed.

That said, you will likely want to utilize extensions in your project.

If you're unsure where to start or what you need you can try the [**@roots/bud-preset-recommend**](/packages/@roots/bud-preset-recommend) preset. Most require zero configuration.

### First-party extensions

There are a number of Roots maintained extensions available to kickstart your project

| Name                                                                                                                         | Project home                                                                                                                                       | Extension docs                                                                                             | Latest                                                                                                       |
| ---------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
| [**@roots/bud-babel**](/packages/@roots/bud-babel)                                   | [**@babel/babel**](https://github.com/babel/babel)                                                                                                 | [ðŸ“š README](/packages/@roots/bud-babel/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-babel.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-compress**](/packages/@roots/bud-compress)                             | [**@webpack-contrib/compression-webpack-plugin**](https://github.com/webpack-contrib/compression-webpack-plugin)                                   | [ðŸ“š README](/packages/@roots/bud-compress/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-compress.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-criticalcss**](/packages/@roots/bud-criticalcss)                       | [**@addyosmani/critical**](https://github.com/addyosmani/critical)                                                                                 | [ðŸ“š README](/packages/@roots/bud-criticalcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-criticalcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-emotion**](/packages/@roots/bud-emotion)                               | [**@emotion/emotion-css**](https://github.com/emotion/emotion-css)                                                                                 | [ðŸ“š README](/packages/@roots/bud-emotion/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-emotion.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-entrypoints**](/packages/@roots/bud-entrypoints)                       | [**@roots/entrypoints-webpack-plugin**](/packages/@roots/entrypoints-webpack-plugin)                       | [ðŸ“š README](/packages/@roots/bud-entrypoints/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-entrypoints.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-esbuild**](/packages/@roots/bud-esbuild)                               | [**@roots/esbuild-loader**](https://github.com/roots/esbuild-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-esbuild/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-esbuild.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-eslint**](/packages/@roots/bud-eslint)                                 | [**@webpack-contrib/eslint-webpack-plugin**](https://github.com/webpack-contrib/eslint-webpack-plugin)                                             | [ðŸ“š README](/packages/@roots/bud-eslint/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-eslint.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-imagemin**](/packages/@roots/bud-imagemin)                             | [**@webpack-contrib/image-minimizer-webpack-plugin**](https://github.com/webpack-contrib/image-minimizer-webpack-plugin)                           | [ðŸ“š README](/packages/@roots/bud-imagemin/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-imagemin.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-library**](/packages/@roots/bud-library)                               | [**@asfktz/autodll-webpack-plugin**](https://github.com/asfktz/autodll-webpack-plugin)                                                             | [ðŸ“š README](/packages/@roots/bud-library/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-library.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-mdx**](/packages/@roots/bud-mdx)                                       | [**@mdx-js/mdx**](https://github.com/mdx-js/mdx)                                                                                                   | [ðŸ“š README](/packages/@roots/bud-mdx/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-mdx.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-postcss**](/packages/@roots/bud-postcss)                               | [**@postcss/postcss**](https://github.com/postcss/postcss)                                                                                         | [ðŸ“š README](/packages/@roots/bud-postcss/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-postcss.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-prettier**](/packages/@roots/bud-prettier)                             | [**@prettier/prettier**](https://github.com/prettier/prettier)                                                                                     | [ðŸ“š README](/packages/@roots/bud-prettier/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-prettier.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-purgecss**](/packages/@roots/bud-purgecss)                             | [**@FullHuman/purgecss**](https://github.com/FullHuman/purgecss)                                                                                   | [ðŸ“š README](/packages/@roots/bud-purgecss/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-purgecss.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-react**](/packages/@roots/bud-react)                                   | [**@facebook/react**](https://github.com/facebook/react)                                                                                           | [ðŸ“š README](/packages/@roots/bud-react/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-react.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-sass**](/packages/@roots/bud-sass)                                     | [**@sass/sass**](https://github.com/sass/sass)                                                                                                     | [ðŸ“š README](/packages/@roots/bud-sass/README.md)                   | ![npm](https://img.shields.io/npm/v/@roots/bud-sass.svg?color=%23525ddc&style=flat-square)                   |
| [**@roots/bud-stylelint**](/packages/@roots/bud-stylelint)                           | [**@stylelint/stylelint**](https://github.com/stylelint/stylelint)                                                                                 | [ðŸ“š README](/packages/@roots/bud-stylelint/README.md)              | ![npm](https://img.shields.io/npm/v/@roots/bud-stylelint.svg?color=%23525ddc&style=flat-square)              |
| [**@roots/bud-tailwindcss**](/packages/@roots/bud-tailwindcss)                       | [**@tailwindlabs/tailwindcss**](https://github.com/tailwindlabs/tailwindcss)                                                                       | [ðŸ“š README](/packages/@roots/bud-tailwindcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-tailwindcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-terser**](/packages/@roots/bud-terser)                                 | [**@terser/terser**](https://github.com/terser/terser)                                                                                             | [ðŸ“š README](/packages/@roots/bud-terser/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-terser.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-typescript**](/packages/@roots/bud-typescript)                         | [**@TypeStrong/ts-loader**](https://github.com/TypeStrong/ts-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-typescript/README.md)             | ![npm](https://img.shields.io/npm/v/@roots/bud-typescript.svg?color=%23525ddc&style=flat-square)             |
| [**@roots/bud-vue**](/packages/@roots/bud-vue)                                       | [**@vue/vue**](https://github.com/vue/vue)                                                                                                         | [ðŸ“š README](/packages/@roots/bud-vue/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-vue.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-wordpress-dependencies**](/packages/@roots/bud-wordpress-dependencies) | [**@roots/wordpress-dependencies-webpack-plugin**](/packages/@roots/wordpress-dependencies-webpack-plugin) | [ðŸ“š README](/packages/@roots/bud-wordpress-dependencies/README.md) | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-dependencies.svg?color=%23525ddc&style=flat-square) |
| [**@roots/bud-wordpress-externals**](/packages/@roots/bud-wordpress-externals)       | [**@roots/wordpress-externals-webpack-plugin**](/packages/@roots/wordpress-externals-webpack-plugin)       | [ðŸ“š README](/packages/@roots/bud-wordpress-externals/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-externals.svg?color=%23525ddc&style=flat-square)    |
| [**@roots/bud-wordpress-manifests**](/packages/@roots/bud-wordpress-manifests)       | [**@roots/merged-manifest-webpack-plugin**](/packages/@roots/merged-manifest-webpack-plugin)               | [ðŸ“š README](/packages/@roots/bud-wordpress-manifests/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-manifests.svg?color=%23525ddc&style=flat-square)    |

### Third-party extensions

Have you produced a Bud extension and want to share it here? Please, create an issue sharing information about your project.
