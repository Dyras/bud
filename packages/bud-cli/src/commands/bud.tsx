import {yargs} from '@roots/bud-support'

import * as build from './build'
import * as publish from './publish'

/**
 * $bud
 */
const cli = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(
    1,
    'You must specify a command. See `bud --help` for usage.\n',
  )
  .usage('\nBud \n\nbud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .epilog('https://github.com/roots/bud').argv

export {cli as bud}
