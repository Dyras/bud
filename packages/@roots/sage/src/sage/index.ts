/**
 * Bud preset: @roots/sage
 */

import {bud, Bud} from '@roots/bud'
import * as babel from '@roots/bud-babel'
import * as esbuild from '@roots/bud-esbuild'
import * as postcss from '@roots/bud-postcss'
import * as react from '@roots/bud-react'
import * as sass from '@roots/bud-sass'
import * as tailwindcss from '@roots/bud-tailwindcss'
import * as typescript from '@roots/bud-typescript'
import * as eslint from '@roots/bud-eslint'
import * as prettier from '@roots/bud-prettier'
import * as stylelint from '@roots/bud-stylelint'
import * as entrypoints from '@roots/bud-entrypoints'
import * as dependencies from '@roots/bud-wordpress-dependencies'
import * as externals from '@roots/bud-wordpress-externals'
import * as manifests from '@roots/bud-wordpress-manifests'
import * as imagemin from '@roots/bud-imagemin'
import cssnano from 'cssnano'

/**
 * Sage - tailwind preset
 */
export const sage = bud
  /**
   * Artifacts/cache store
   *
   * Set to Acorn standard location
   */
  .storage('storage/bud')

  /**
   * Src path
   */
  .srcPath('resources')

  /**
   * Dist path
   */
  .distPath('public')

  /**
   * Public path
   */
  .when(
    ({env}: Bud) => env.has('APP_PUBLIC_PATH'),
    ({publicPath, env}: Bud) =>
      publicPath(env.get('APP_PUBLIC_PATH')),
  )

  /**
   * ESBuild doesn't support HMR. it is purely a transpiler.
   *
   * - snowpack & vite each have their own HMR solutions.
   * - snowpack provides their solution as an
   *   [ESM spec proposal](https://github.com/snowpackjs/esm-hmr)
   * - [Related article from Dan Abramov](https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf)
   *
   * Losing hmr in development is not worth the theoretical
   * lost time in dev, for most users.
   *
   * Thus, Sage uses esbuild in production, and babel/hmr in development.
   */
  .when(
    ({isDevelopment}) => isDevelopment,
    ({use}: Bud) => use(typescript).use(babel).use(react),
    ({use}: Bud) => use(esbuild).esbuild.jsx(),
  )

  /**
   * General extensions
   */
  .use([
    /**
     * Style transpilation
     */
    postcss,
    sass,
    tailwindcss,

    /**
     * Linting
     */
    eslint,
    prettier,
    stylelint,

    /**
     * Manifests
     */
    entrypoints,
    dependencies,
    externals,
    manifests,
  ])

  /**
   * Sage webpack aliases
   */
  .alias({
    '@fonts': 'fonts',
    '@images': 'images',
    '@scripts': 'scripts',
    '@styles': 'styles',
  })

  /**
   * Provide
   */
  .provide({
    jquery: ['$', 'jQuery'],
  })

  /**
   * Additional options
   */
  .when(
    ({isProduction}) => isProduction,
    ({sequence}) =>
      sequence([
        ({use}) => use(imagemin),
        ({postcss}) =>
          postcss.setPlugin([
            'cssnano',
            cssnano({preset: 'default'}),
          ]),
        (sage: Bud) => sage.minify().hash().vendor().runtime(),
      ]),
    ({sequence}) =>
      sequence([
        ({env, proxy}) =>
          env.has('APP_PROXY_HOST') &&
          proxy({host: env.get('APP_PROXY_HOST')}),
        ({env, proxy}) =>
          env.has('APP_PROXY_PORT') &&
          proxy({port: sage.env.get('APP_PROXY_PORT')}),
      ]),
  )
