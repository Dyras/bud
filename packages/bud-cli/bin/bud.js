#!/usr/bin/env node

require('../lib/cjs/commands/bud')
