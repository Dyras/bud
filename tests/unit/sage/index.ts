import '@roots/bud-api'

import {config, factory, Framework} from '@roots/bud'
import * as Sage from '@roots/sage'

describe('@roots/sage', () => {
  let SAGE_DIR = process.cwd().concat('/examples/sage')

  let SAGE_CFG = {
    config: {
      ...config,

      location: {
        ...config.location,
        project: SAGE_DIR,
      },
    },
  }

  let bud: Framework = null
  let bootSpy = jest.spyOn(Sage, 'boot')

  beforeAll(() => {
    bud = factory(SAGE_CFG)
    bud.use(Sage)
  })

  afterAll(done => {
    bud.close(done)
  })

  it('extension has name prop', () => {
    expect(Sage.name).toBe('@roots/sage')
  })

  it('extension boot method was called', () => {
    expect(bootSpy).toHaveBeenCalled()
  })

  it(`sage required peers matches snapshot`, () => {
    expect(bud.project.get('peers')).toMatchSnapshot({
      '@wordpress/browserslist-config': {
        name: '@wordpress/browserslist-config',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      eslint: {
        name: 'eslint',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      postcss: {
        name: 'postcss',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      'postcss-import': {
        name: 'postcss-import',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      'postcss-preset-env': {
        name: 'postcss-preset-env',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      react: {
        name: 'react',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      'react-dom': {
        name: 'react-dom',
        type: 'devDependencies',
        ver: expect.any(String),
      },
      tailwindcss: {
        name: 'tailwindcss',
        type: 'devDependencies',
        ver: expect.any(String),
      },
    })
  })

  it('sage resolveFrom matches snapshot', () => {
    expect(bud.project.resolveFrom).toMatchSnapshot([
      expect.stringContaining('@roots/bud'),
      expect.stringContaining('@roots/sage'),
      expect.stringContaining('@roots/bud-preset-wordpress'),
      expect.stringContaining('@roots/bud-preset-recommend'),
      expect.stringContaining('@roots/bud-babel'),
      expect.stringContaining('@roots/bud-entrypoints'),
      expect.stringContaining('@roots/bud-postcss'),
      expect.stringContaining('@roots/bud-react'),
      expect.stringContaining('@roots/bud-wordpress-externals'),
      expect.stringContaining(
        '@roots/bud-wordpress-dependencies',
      ),
      expect.stringContaining('@roots/bud-wordpress-manifests'),
      expect.stringContaining('@roots/bud-eslint'),
      expect.stringContaining('@roots/bud-prettier'),
      expect.stringContaining('@roots/bud-stylelint'),
      expect.stringContaining('@roots/bud-tailwindcss'),
    ])
  })

  it('has expected paths', () => {
    expect(bud.path('storage')).toEqual(
      SAGE_DIR.concat('/storage/bud'),
    )
    expect(bud.publicPath()).toEqual('')
    expect(bud.path('dist')).toEqual(SAGE_DIR.concat('/public'))
    expect(bud.path('src')).toEqual(
      SAGE_DIR.concat('/resources'),
    )
  })

  it('has expected aliases', () => {
    bud.build.make()
    const {alias: aliases} = bud.build.config.resolve

    expect(aliases['@fonts']).toEqual(
      SAGE_DIR.concat('/resources/fonts'),
    )
    expect(aliases['@images']).toEqual(
      SAGE_DIR.concat('/resources/images'),
    )
    expect(aliases['@scripts']).toEqual(
      SAGE_DIR.concat('/resources/scripts'),
    )
    expect(aliases['@styles']).toEqual(
      SAGE_DIR.concat('/resources/styles'),
    )
  })

  it('babel is used', () => {
    expect(bud.extensions.has('@roots/bud-babel')).toEqual(true)
  })

  it('postcss is used', () => {
    expect(bud.extensions.has('@roots/bud-postcss')).toEqual(
      true,
    )
  })

  it('react is used', () => {
    expect(bud.extensions.has('@roots/bud-react')).toEqual(true)
  })

  it('tailwindcss is used', () => {
    expect(bud.extensions.has('@roots/bud-tailwindcss')).toEqual(
      true,
    )
  })
})
