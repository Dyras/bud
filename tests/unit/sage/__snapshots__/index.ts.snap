// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@roots/sage sage required peers matches snapshot 1`] = `
Object {
  "@wordpress/browserslist-config": Object {
    "name": "@wordpress/browserslist-config",
    "version": "4.1.0",
  },
  "eslint": Object {
    "name": "eslint",
    "version": "7.32.0",
  },
  "postcss": Object {
    "name": "postcss",
    "version": "8.3.11",
  },
  "postcss-import": Object {
    "name": "postcss-import",
    "version": "14.0.2",
  },
  "postcss-preset-env": Object {
    "name": "postcss-preset-env",
    "version": "6.7.0",
  },
  "prettier": Object {
    "name": "prettier",
    "version": "2.4.1",
  },
  "react": Object {
    "name": "react",
    "version": "17.0.2",
  },
  "react-dom": Object {
    "name": "react-dom",
    "version": "17.0.2",
  },
  "tailwindcss": Object {
    "name": "tailwindcss",
    "version": "^2.2.8",
  },
}
`;

exports[`@roots/sage sage resolveFrom matches snapshot 1`] = `
Array [
  StringContaining "resources",
  StringContaining "node_modules",
  StringContaining "@roots/sage",
  StringContaining "@roots/bud-preset-wordpress",
  StringContaining "@roots/bud-preset-recommend",
  StringContaining "@roots/bud-babel",
  StringContaining "@roots/bud-entrypoints",
  StringContaining "@roots/bud-postcss",
  StringContaining "@roots/bud-react",
  StringContaining "@roots/bud-wordpress-externals",
  StringContaining "@roots/bud-wordpress-dependencies",
  StringContaining "@roots/bud-wordpress-manifests",
  StringContaining "@roots/bud-eslint",
  StringContaining "@roots/bud-prettier",
  StringContaining "@roots/bud-stylelint",
  StringContaining "@roots/bud-tailwindcss",
  StringContaining "node_modules",
]
`;
